src/test/java/org/jenkinsci/plugins/workflow/steps/ArtifactArchiverStepTest.java

```diff
@@ -74,5 +78,18 @@ public void archive() throws Exception {
         j.assertLogContains("one/two", b);
     }
 
-}
+    @Issue("JENKINS-49635")
+    @Test
+    public void directDownload() throws Exception {
+        ArtifactManagerConfiguration.get().getArtifactManagerFactories().add(new DirectArtifactManagerFactory());
+        j.createSlave("remote1", null, null);
+        j.createSlave("remote2", null, null);
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node('remote1') {writeFile file: 'x', text: 'contents'; archiveArtifacts 'x'}; node('remote2') {unarchive mapping: [x: 'x']; echo(/loaded ${readFile('x')}/)}", true));
```

Comment:
no `stash`/`unstash` test?

Reply:
No, because that behavior is not being defined in this PR—it would be overridden in the plugin providing the `ArtifactManagerFactory`.